### ISSUE TITLE
Wrong solution file detected

### ISSUE THREAD

**Issue by depsir** [exercism/javascript-analyzer#2]
I'm trying to run the analyzer against solution a1cac61696d1428c8b55bdd8dc39cfa0.

The downloaded folder contains a file `babel.config.js` which is detected as solution file and the analyzer complains that the right `export` is missing:

```json
=> early finialization (disapprove_with_comment)
=> output:

{
  "status": "disapprove_with_comment",
  "comments": [
    {
      "comment": "No method called `%<method_name>s`. The tests won't pass without it.",
      "params": {
        "method_name": "twoFer"
      }
    },
    {
      "comment": "No [export](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) called `%<export_name>s`.\nThe tests won't pass without it.\n\nDid you forget adding: `export %<export_name>s`?",
      "params": {
        "export_name": "twoFer"
      }
    }
  ]
}
```

If I delete that file, everything works as expected:
```json
=> early finialization (approve_as_optimal)
=> output:

{
  "status": "approve_as_optimal",
  "comments": []
}
```

**Comment by SleeplessByte** [comment]
@depsir cool! 

[This code](https://github.com/exercism/javascript-analyzer/blob/master/src/solution.ts#L22) is supposed to take care of that:
```javascript
const preferredMatches = preferredNames
  ? candidates.filter(file => preferredNames.includes(file))
  : []
``` 

Is the actual file not called `two-fer.js` by any chance? 

We can make this _more robust_ by adding another filter [here](https://github.com/exercism/javascript-analyzer/blob/master/src/solution.ts#L20), excluding _all_ the config files, such as this one.

**Comment by depsir** [comment]
Is it possible that the problem is in the `preferredNames` [definition?](https://github.com/exercism/javascript-analyzer/blob/79d46164ea3ecaf70e4fc28fa649f2df36c90cdf/src/solution.ts#L58)

I see a `ts` extension in the code, so this is why, in my opinion, does not choose `two-fer.js`

**Comment by SleeplessByte** [comment]
You're absolutely correct. I wrote the `ts` analyzer first and forgot to update that. Would you like to submit a PR?
